### ISSUE TITLE
Unable to extract uploader id

### ISSUE THREAD

**Issue by profcalculus** [ytdl-org/youtube-dl#32031]
Using latest youtube-dl 2021.12.17 (manually installed) on Ubuntu 22.04.2 LTS . I've tried several YouTube videos, all with the same results. For example:
```
youtube-dl -v https://www.youtube.com/watch?v=3hz6Tb1i2FY
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.youtube.com/watch?v=3hz6Tb1i2FY']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.10.6 (CPython) - Linux-5.19.0-38-generic-x86_64-with-glibc2.35
[debug] exe versions: ffmpeg 4.4.2, ffprobe present, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] 3hz6Tb1i2FY: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1794, in _real_extract
    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

I know this issue has been logged (and reported fixed) before, but youtube-dl is not working for me.
Thanks!

**Comment by dirkf** [comment]
#30839 !

What happened when you tried to update the program using one of the methods recommended in the master issue linked from #30839? Report there if the update didn't work. You should probably replace your version with an unofficial nightly build. If you installed using a package manager, either uninstall the program first, or use the _pip_ update method instead.

**Comment by LastStardaughter** [comment]
I've been having the same problem for weeks! I figured youtube-dl was abandoned or something!

ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 815, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 836, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\youtube.py", line 1794, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 1012, in _search_regex
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Comment by LastStardaughter** [comment]
Followed the trail and switched to yt-dlp for now. If a critical issue like this has been fixed, why no new release though? Doesn't it kind of make youtube-dl unusable?

**Comment by BirgitVeraSchmidt** [comment]
> What happened when you tried to update the program using one of the methods recommended in the master issue linked from https://github.com/ytdl-org/youtube-dl/issues/30839?

The word "update" does not appear a single time in that link.

A more helpful link to fix the problem locally is #31530 , in particular [this comment](https://github.com/ytdl-org/youtube-dl/issues/31530#issuecomment-1435975611).

The key observation is that a nightly build is needed, and that YouTube might be doing some kind of A/B-testing where you sometimes see one format, sometimes the other.

I got it to work with these commands, using a nightly build:
```
sudo wget https://github.com/ytdl-patched/youtube-dl/releases/download/2023.04.12.114514/youtube-dl -O /usr/local/bin/youtube-dl
sudo chmod a+rx /usr/local/bin/youtube-dl
hash -r
```

**Comment by dirkf** [comment]
You found a known issue that has a resolution (albeit incomplete today).

As you found, that issue is listed in the pinned issue that you were asked to to read and follow. It would also have shown up close the top of the recommended issue search.

>...YouTube might be doing some kind of A/B-testing where you sometimes see one format, sometimes the other.

The new owner format has always been present as far as we know, for the last ~2 months, except for some music videos.

>I got it to work with these commands, using a nightly build:

Like installing the release build apart from the URL. `hash -r` shouldn't be necessary if you're using the same pathname for the new yt-dl (but POSIX wording on this is obscure).

>... why no new release ...

#31585.

**Comment by BirgitVeraSchmidt** [comment]
Sorry for the slightly peeved tone in my earlier reply.

I *did* come here through the issue search (I'm not the original author), only found a reply that I misinterpreted for a simple "read this before posting" and "have you tried updating?", went to put together the information from other links, and decided to add an answer here that might be helpful to the next person coming across this via the issue search.

I realized only after posting (see my edit) that I had misread "update the program using one of the methods recommended in the master issue linked from https://github.com/ytdl-org/youtube-dl/issues/30839" as "go to the master issue https://github.com/ytdl-org/youtube-dl/issues/30839 and update the program using one of the methods linked there".

So, sorry about that.

**Comment by fabtjar** [comment]
> > What happened when you tried to update the program using one of the methods recommended in the master issue linked from #30839?
> 
> The word "update" does not appear a single time in that link, (edited:) though upon fifth reading I finally understood what was meant by "master issue linked from".
> 
> A more helpful link to fix the problem locally is #31530 , in particular [this comment](https://github.com/ytdl-org/youtube-dl/issues/31530#issuecomment-1435975611).
> 
> The key observation is that a nightly build is needed, and that YouTube might be doing some kind of A/B-testing where you sometimes see one format, sometimes the other.
> 
> I got it to work with these commands, using a nightly build:
> 
> ```
> sudo wget https://github.com/ytdl-patched/youtube-dl/releases/download/2023.04.12.114514/youtube-dl -O /usr/local/bin/youtube-dl
> sudo chmod a+rx /usr/local/bin/youtube-dl
> hash -r
> ```

this works for me!
(I used `/usr/bin/youtube-dl` instead as dnf installed it there originally)
